zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplineCompiler.kt

```diff
@@ -77,10 +78,32 @@ object ZiplineCompiler {
         }
       }
     }
-    val manifest = ZiplineManifest.create(modules)
+
+    val sortedModules = ZiplineManifest.create("temp", "temp", modules).modules
+    val applicationModuleJsFileNameWithoutExtention = sortedModules.entries.last().key.removePrefix("./").removeSuffix(".js")
+    val outputTsTypeDefinitionFile = File(inputDir.path, "${applicationModuleJsFileNameWithoutExtention}.d.ts")
+    val applicationId = "./${applicationModuleJsFileNameWithoutExtention}.js"
+    val prepareFunction = getPrepareFunctionName(outputTsTypeDefinitionFile)
+
+    val manifest = ZiplineManifest.create(
+      applicationId = applicationId,
+      prepareFunction = prepareFunction,
+      modules = modules,
+    )
     val manifestFile = File(outputDir.path, "manifest.zipline.json")
     manifestFile.writeText(Json.encodeToString(manifest))
   }
+
+  internal fun getPrepareFunctionName(tsTypeDefinitionFile: File): String {
+    val contents = tsTypeDefinitionFile.readText()
+    return getPrepareFunctionName(contents)
+  }
+
+  internal fun getPrepareFunctionName(contents: String): String {
+    val jsPackages = "(?<=(export namespace\\s)).*(?=\\s\\{)".toRegex().findAll(contents).toList().map { it.value }.first()
+    val functionName = "(?<=(function\\s)).*(?=\\(\\))".toRegex().findAll(contents).toList().map { it.value }.first()
```

Comment:
Rather than `.first()` I’d prefer `.single()` since it explodes better if it’s ambiguous

Reply:
Perfect, I was looking for that API but forgot what it was called